<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Curated Metagenomic Data of the Human Microbiome â€¢ curatedMetagenomicData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Curated Metagenomic Data of the Human Microbiome">
<meta property="og:description" content="
    The curatedMetagenomicData package provides standardized, curated human
    microbiome data for novel analyses. It includes gene families, marker
    abundance, marker presence, pathway abundance, pathway coverage, and
    relative abundance for samples collected from different body sites. The
    bacterial, fungal, and archaeal taxonomic abundances for each sample were
    calculated with MetaPhlAn3, and metabolic functional potential was
    calculated with HUMAnN3. The manually curated sample metadata and
    standardized metagenomic data are available as (Tree)SummarizedExperiment
    objects.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">curatedMetagenomicData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.4.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/curatedMetagenomicData.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/available-studies.html">Available Studies</a>
    </li>
    <li>
      <a href="articles/our-pipeline.html">Our Pipeline</a>
    </li>
    <li>
      <a href="articles/version-three.html">Version Three</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/curatedMetagenomicData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="curatedmetagenomicdata">curatedMetagenomicData<a class="anchor" aria-label="anchor" href="#curatedmetagenomicdata"></a>
</h1></div>
<!-- badges: start -->

<p>The <em><a href="https://bioconductor.org/packages/3.15/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em> package provides standardized, curated human microbiome data for novel analyses. It includes gene families, marker abundance, marker presence, pathway abundance, pathway coverage, and relative abundance for samples collected from different body sites. The bacterial, fungal, and archaeal taxonomic abundances for each sample were calculated with <a href="https://github.com/biobakery/MetaPhlAn" class="external-link">MetaPhlAn3</a>, and metabolic functional potential was calculated with <a href="https://github.com/biobakery/humann" class="external-link">HUMAnN3</a>. The manually curated sample metadata and standardized metagenomic data are available as (Tree)SummarizedExperiment objects.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install <em><a href="https://bioconductor.org/packages/3.15/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em> from Bioconductor, use <em><a href="https://CRAN.R-project.org/package=BiocManager" class="external-link">BiocManager</a></em> as follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"curatedMetagenomicData"</span><span class="op">)</span></code></pre></div>
<p>To install <em><a href="https://bioconductor.org/packages/3.15/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em> from GitHub, use <em><a href="https://CRAN.R-project.org/package=BiocManager" class="external-link">BiocManager</a></em> as follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"waldronlab/curatedMetagenomicData"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Most users should simply install <em><a href="https://bioconductor.org/packages/3.15/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em> from Bioconductor.</p>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>To access curated metagenomic data, users will use the <code><a href="reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code> method both to query and return resources. Multiple resources can be queried or returned with a single call to <code><a href="reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code>, but only the titles of resources are returned by default.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"AsnicarF_20.+"</span><span class="op">)</span>
<span class="co">## 2021-03-31.AsnicarF_2017.gene_families</span>
<span class="co">## 2021-03-31.AsnicarF_2017.marker_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2017.marker_presence</span>
<span class="co">## 2021-03-31.AsnicarF_2017.pathway_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2017.pathway_coverage</span>
<span class="co">## 2021-03-31.AsnicarF_2017.relative_abundance</span>
<span class="co">## 2021-10-14.AsnicarF_2017.gene_families</span>
<span class="co">## 2021-10-14.AsnicarF_2017.marker_abundance</span>
<span class="co">## 2021-10-14.AsnicarF_2017.marker_presence</span>
<span class="co">## 2021-10-14.AsnicarF_2017.pathway_abundance</span>
<span class="co">## 2021-10-14.AsnicarF_2017.pathway_coverage</span>
<span class="co">## 2021-10-14.AsnicarF_2017.relative_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.gene_families</span>
<span class="co">## 2021-03-31.AsnicarF_2021.marker_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.marker_presence</span>
<span class="co">## 2021-03-31.AsnicarF_2021.pathway_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.pathway_coverage</span>
<span class="co">## 2021-03-31.AsnicarF_2021.relative_abundance</span></code></pre></div>
<p>When the <code>dryrun</code> argument is set to <code>FALSE</code>, a <code>list</code> of <code>SummarizedExperiment</code> and/or <code>TreeSummarizedExperiment</code> objects is returned. The <code>rownames</code> argument determines the type of <code>rownames</code> to use for <code>relative_abundance</code> resources: either <code>"long"</code> (the default), <code>"short"</code> (species name), or <code>"NCBI"</code> (NCBI Taxonomy ID). When a single resource is requested, a single element <code>list</code> is returned.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"AsnicarF_2017.relative_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span>, rownames <span class="op">=</span> <span class="st">"short"</span><span class="op">)</span>
<span class="co">## $`2021-10-14.AsnicarF_2017.relative_abundance`</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 296 24 </span>
<span class="co">## metadata(1): agglomerated_by_rank</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(296): Escherichia coli Bifidobacterium bifidum ...</span>
<span class="co">##   Streptococcus gordonii Abiotrophia sp. HMSC24B09</span>
<span class="co">## rowData names(7): superkingdom phylum ... genus species</span>
<span class="co">## colnames(24): MV_FEI1_t1Q14 MV_FEI2_t1Q14 ... MV_MIM5_t2M14</span>
<span class="co">##   MV_MIM5_t3F15</span>
<span class="co">## colData names(22): study_name subject_id ... lactating curator</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (296 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span></code></pre></div>
<p>When the <code>counts</code> argument is set to <code>TRUE</code>, relative abundance proportions are multiplied by read depth and rounded to the nearest integer prior to being returned. Also, when multiple resources are requested, the <code>list</code> will contain named elements corresponding to each <code>SummarizedExperiment</code> and/or <code>TreeSummarizedExperiment</code> object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"AsnicarF_20.+.relative_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span>, counts <span class="op">=</span> <span class="cn">TRUE</span>, rownames <span class="op">=</span> <span class="st">"short"</span><span class="op">)</span>
<span class="co">## $`2021-10-14.AsnicarF_2017.relative_abundance`</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 296 24 </span>
<span class="co">## metadata(1): agglomerated_by_rank</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(296): Escherichia coli Bifidobacterium bifidum ...</span>
<span class="co">##   Streptococcus gordonii Abiotrophia sp. HMSC24B09</span>
<span class="co">## rowData names(7): superkingdom phylum ... genus species</span>
<span class="co">## colnames(24): MV_FEI1_t1Q14 MV_FEI2_t1Q14 ... MV_MIM5_t2M14</span>
<span class="co">##   MV_MIM5_t3F15</span>
<span class="co">## colData names(22): study_name subject_id ... lactating curator</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (296 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span>
<span class="co">## </span>
<span class="co">## $`2021-03-31.AsnicarF_2021.relative_abundance`</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 633 1098 </span>
<span class="co">## metadata(1): agglomerated_by_rank</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(633): Phocaeicola vulgatus Bacteroides stercoris ...</span>
<span class="co">##   Pyramidobacter sp. C12-8 Brevibacterium aurantiacum</span>
<span class="co">## rowData names(7): superkingdom phylum ... genus species</span>
<span class="co">## colnames(1098): SAMEA7041133 SAMEA7041134 ... SAMEA7045952 SAMEA7045953</span>
<span class="co">## colData names(22): study_name subject_id ... BMI family</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (633 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="analyses">Analyses<a class="anchor" aria-label="anchor" href="#analyses"></a>
</h2>
<p>See <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses" class="external-link">curatedMetagenomicAnalyses</a> for analyses in R and Python using <em><a href="https://bioconductor.org/packages/3.15/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em>.</p>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<p>To contribute to the <em><a href="https://bioconductor.org/packages/3.15/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></em> R/Bioconductor package, first read the <a href="CONTRIBUTING.html">contributing guidelines</a> and then open an issue. Also, note that in contributing you agree to abide by the <a href="CODE_OF_CONDUCT.html">code of conduct</a>.</p>
<hr>
<p>Pasolli E, Schiffer L, Manghi P, Renson A, Obenchain V, Truong D, Beghini F, Malik F, Ramos M, Dowd J, Huttenhower C, Morgan M, Segata N, Waldron L (2017). Accessible, curated metagenomic data through ExperimentHub. <em>Nat. Methods</em>, <strong>14</strong> (11), 1023-1024. ISSN 1548-7091, 1548-7105, doi: <a href="https://doi.org/10.1038/nmeth.4468" class="external-link">10.1038/nmeth.4468</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/curatedMetagenomicData" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/waldronlab/curatedMetagenomicData/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/waldronlab/curatedMetagenomicData/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing curatedMetagenomicData</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Lucas schiffer <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-3628-0326" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Levi Waldron <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.codefactor.io/repository/github/waldronlab/curatedmetagenomicdata" class="external-link"><img src="https://img.shields.io/codefactor/grade/github/waldronlab/curatedMetagenomicData" alt="code quality"></a></li>
<li><a href="https://codecov.io/gh/waldronlab/curatedMetagenomicData" class="external-link"><img src="https://img.shields.io/codecov/c/github/waldronlab/curatedMetagenomicData" alt="coverage"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Lucas schiffer, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
