---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
library(curatedMetagenomicData)
```

# curatedMetagenomicData

<!-- badges: start -->
[![code quality](https://img.shields.io/codefactor/grade/github/waldronlab/curatedMetagenomicData)](https://www.codefactor.io/repository/github/waldronlab/curatedmetagenomicdata)
[![coverage](https://img.shields.io/codecov/c/github/waldronlab/curatedMetagenomicData)](https://codecov.io/gh/waldronlab/curatedMetagenomicData)
<!-- badges: end -->

The `r BiocStyle::Biocpkg("curatedMetagenomicData")` package provides standardized, curated human microbiome data for novel analyses. It includes gene families, marker abundance, marker presence, pathway abundance, pathway coverage, and relative abundance for samples collected from different body sites. The bacterial, fungal, and archaeal taxonomic abundances for each sample were calculated with [MetaPhlAn3](https://github.com/biobakery/MetaPhlAn), and metabolic functional potential was calculated with [HUMAnN3](https://github.com/biobakery/humann). The manually curated sample metadata and standardized metagenomic data are available as (Tree)SummarizedExperiment objects.

## Installation

To install `r BiocStyle::Biocpkg("curatedMetagenomicData")` from Bioconductor, use `r BiocStyle::CRANpkg("BiocManager")` as follows.

```{r, eval = FALSE}
BiocManager::install("curatedMetagenomicData")
```

To install `r BiocStyle::Biocpkg("curatedMetagenomicData")` from GitHub, use `r BiocStyle::CRANpkg("BiocManager")` as follows.

```{r, eval = FALSE}
BiocManager::install("waldronlab/curatedMetagenomicData", dependencies = TRUE, build_vignettes = TRUE)
```

Most users should simply install `r BiocStyle::Biocpkg("curatedMetagenomicData")` from Bioconductor.

## Examples

To access curated metagenomic data, users will use the `curatedMetagenomicData()` method both to query and return resources. Multiple resources can be queried or returned with a single call to `curatedMetagenomicData()`, but only the titles of resources are returned by default.

```{r, collapse = TRUE}
curatedMetagenomicData("AsnicarF_20.+")
```

When the `dryrun` argument is set to `FALSE`, a `list` of `SummarizedExperiment` and/or `TreeSummarizedExperiment` objects is returned. The `rownames` argument determines the type of `rownames` to use for `relative_abundance` resources: either `"long"` (the default), `"short"` (species name), or `"NCBI"` (NCBI Taxonomy ID). When a single resource is requested, a single element `list` is returned.

```{r, collapse = TRUE, message = FALSE}
curatedMetagenomicData("AsnicarF_2017.relative_abundance", dryrun = FALSE, rownames = "short")
```

When the `counts` argument is set to `TRUE`, relative abundance proportions are multiplied by read depth and rounded to the nearest integer prior to being returned. Also, when multiple resources are requested, the `list` will contain named elements corresponding to each `SummarizedExperiment` and/or `TreeSummarizedExperiment` object.

```{r, collapse = TRUE, message = FALSE}
curatedMetagenomicData("AsnicarF_20.+.relative_abundance", dryrun = FALSE, counts = TRUE, rownames = "short")
```

## Analyses

See [curatedMetagenomicAnalyses](https://github.com/waldronlab/curatedMetagenomicAnalyses) for analyses in R and Python using `r BiocStyle::Biocpkg("curatedMetagenomicData")`.

## Contributing

To contribute to the `r BiocStyle::Biocpkg("curatedMetagenomicData")` R/Bioconductor package, first read the [contributing guidelines](CONTRIBUTING.md) and then open an issue. Also, note that in contributing you agree to abide by the [code of conduct](CODE_OF_CONDUCT.md).

---

Pasolli E, Schiffer L, Manghi P, Renson A, Obenchain V, Truong D, Beghini F, Malik F, Ramos M, Dowd J, Huttenhower C, Morgan M, Segata N, Waldron L (2017). Accessible, curated metagenomic data through ExperimentHub. *Nat. Methods*, **14** (11), 1023-1024. ISSN 1548-7091, 1548-7105, doi: [10.1038/nmeth.4468](https://doi.org/10.1038/nmeth.4468).
