---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

# curatedMetagenomicData

<!-- badges: start -->
[![code quality](https://img.shields.io/codefactor/grade/github/waldronlab/curatedMetagenomicData)](https://www.codefactor.io/repository/github/waldronlab/curatedmetagenomicdata)
[![coverage](https://img.shields.io/codecov/c/github/waldronlab/curatedMetagenomicData)](https://codecov.io/gh/waldronlab/curatedMetagenomicData)
<!-- badges: end -->

The curatedMetagenomicData package provides standardized, curated human microbiome data for novel analyses. It includes gene families, marker abundance, marker presence, pathway abundance, pathway coverage, and relative abundance for samples collected from different body sites. The bacterial, fungal, and archaeal taxonomic abundances for each sample were calculated with MetaPhlAn3 and metabolic functional potential was calculated with HUMAnN3. The manually curated sample metadata and standardized metagenomic data are available as (Tree)SummarizedExperiment objects.

## Installation

Most users should simply install the current version of `r BiocStyle::Biocpkg("curatedMetagenomicData")` from Bioconductor, unless they have good reason not to. Changes to the package are always committed to GitHub first, and it might be desirable to have the absolute latest changes by installing `r BiocStyle::Githubpkg("waldronlab/curatedMetagenomicData")` from GitHub; although, this is not recommended for most users. Finally, we have made extensive changes in `r BiocStyle::Biocpkg("curatedMetagenomicData")` `3.0.0` and some users might want to revert to the previous version of `r BiocStyle::Githubpkg("waldronlab/curatedMetagenomicData")` from GitHub for older analyses. The previous version of `r BiocStyle::Githubpkg("waldronlab/curatedMetagenomicData")` from GitHub, `1.20.0`, will remain installable for the foreseeable future.

To install the current version of `r BiocStyle::Biocpkg("curatedMetagenomicData")` from Bioconductor, use `r BiocStyle::CRANpkg("BiocManager")` as follows.

```{r, eval = FALSE}
BiocManager::install("curatedMetagenomicData")
```

To install the current version of `r BiocStyle::Githubpkg("waldronlab/curatedMetagenomicData")` from GitHub, use `r BiocStyle::CRANpkg("BiocManager")` as follows.

```{r, eval = FALSE}
BiocManager::install("waldronlab/curatedMetagenomicData", dependencies = TRUE, build_vignettes = TRUE)
```

To install the previous version of `r BiocStyle::Githubpkg("waldronlab/curatedMetagenomicData")` from GitHub, use `r BiocStyle::CRANpkg("BiocManager")` as follows.

```{r, eval = FALSE}
BiocManager::install("waldronlab/curatedMetagenomicData", dependencies = TRUE, build_vignettes = TRUE, ref = "v1.20.0")
```

## Examples

```{r, include = FALSE}
library(curatedMetagenomicData)
```

To access curated metagenomic data, users will use the `curatedMetagenomicData()` method both to query and return resources. Multiple resources can be queried or returned with a single call to `curatedMetagenomicData()`, but only the titles of resources are returned by default.

```{r, collapse = TRUE}
curatedMetagenomicData("AsnicarF_20.+")
```

When the second argument `dryrun` is set to `FALSE`, a `list` of `SummarizedExperiment` and/or `TreeSummarizedExperiment` objects is returned. When a single resource is requested, a single element `list` is returned.

```{r, collapse = TRUE, message = FALSE}
curatedMetagenomicData("AsnicarF_2017.relative_abundance", dryrun = FALSE)
```

When the third argument `counts` is set to `TRUE`, relative abundance proportions are multiplied by read depth and rounded to the nearest integer prior to being returned. Also, when multiple resources are requested, the `list` will contain named elements corresponding to each `SummarizedExperiment` and/or `TreeSummarizedExperiment` object.

```{r, collapse = TRUE, message = FALSE}
curatedMetagenomicData("AsnicarF_20.+.relative_abundance", dryrun = FALSE, counts = TRUE)
```

### Example Analyses

See [curatedMetagenomicAnalyses](https://github.com/waldronlab/curatedMetagenomicAnalyses)
for analyses in R and Python using curatedMetagenomicData.

## Contributing

All are welcome to contribute to the curatedMetagenomicData package provided contributions are appropriate. Please see the contributing guide.

## Code of Conduct
  
Please note that the curatedMetagenomicData package is released with a code of conduct. By contributing, you agree to abide by its terms.

---

Pasolli E, Schiffer L, Manghi P, Renson A, Obenchain V, Truong D, Beghini F, Malik F, Ramos M, Dowd J, Huttenhower C, Morgan M, Segata N, Waldron L (2017). Accessible, curated metagenomic data through ExperimentHub. *Nat. Methods*, **14** (11), 1023-1024. ISSN 1548-7091, 1548-7105, doi: [10.1038/nmeth.4468](https://doi.org/10.1038/nmeth.4468).
