---
title: "curatedMetagenomicData"
abstract: >
    The curatedMetagenomicData package provides taxonomic, functional, and gene
    marker abundance for samples collected from different bodysites. It provides
    data from aproximately 3000 human microbiome samples that have been highly
    processed, refined, and curated such that analysis that might otherwise
    require a computing cluster can be done on an ordianary laptop.
output:
    BiocStyle::html_document
vignette: >
    %\VignetteIndexEntry{curatedMetagenomicData}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# What `curatedMetagenomicData` provides

`curatedMetagenomicData` provides 6 types of data for each dataset:

1. species-level taxonomic profiles,
2. presence of unique, clade-specific markers,
3. abundance of unique, clade-specific markers,
4. abundance of gene families,
5. metabolic pathway coverage, and 
6. metabolic pathway abundance. 

Types 1-3 are generated by 
[MetaPhlAn2](http://huttenhower.sph.harvard.edu/metaphlan2); 4-6 are generated 
by [HUMAnN2](http://huttenhower.sph.harvard.edu/humann2).

Currently, `curatedMetagenomicData` provides:

* 2,875 samples from 13 datasets, primarily of the human gut but including body 
sites profiled in the Human Microbiome Project
* processed data from whole-metagenome shotgun metagenomics, with 
manually-curated metadata, as integrated and documented Bioconductor 
ExpressionSet objects
* ~200 fields of specimen metadata from original papers, supplementary files, 
and websites, with manual curation to standardize annotations
* processing of data through the new MetaPhlAn2 pipeline 
(http://huttenhower.sph.harvard.edu/metaphlan2) for taxonomic abundance, and 
HUMAnN2 pipeline (Version 2 manuscript in preparation) 
(http://huttenhower.sph.harvard.edu/humann2) for metabolic analysis.
* This effort required analyzing ~23T of raw sequencing data, generating ~52T of
intermediate files

These datasets are documented in the [reference manual]().

# Using `curatedMetagenomicData` Resources

Use of the resources in `curatedMetagenomicData` is simplified with the use of
Bioconductor's ExperimentHub platform, which allows for the accessing of data
through an intuitive interface. First, `curatedMetagenomicData` is installed
using `BiocInstaller` and then called as a library - the process allows for the
user to simply call datasets as functions because the package is aware of the
resources present in ExperimentHub S3 buckets. A function call to a dataset name
will return an expression set object which can be manipulated or serialized.

```{r, eval=FALSE}
BiocInstaller::biocLite("curatedMetagenomicData")
library(curatedMetagenomicData)
LomanNJ_2013_Hi.genefamilies_relab.stool()
```

# Features of `ExpressionSet` Objects

All datasets are represented as `ExpressionSet` objects because of the
integrative nature of the class and its ability to bind data and metadata. There
are three main functions, from the `Biobase` package, that provide access to
experiment-level metadata, subject-level metadata, and the data itself.

To access the experiment-level metadata the `experimentData()` function is used
to return a `MIAME` (Minimum Information About a Microarray Experiment) object.

```{r, eval=FALSE}
LomanNJ_2013_Hi.genefamilies_relab.stool() %>%
experimentData()
```

To access the subject-level metadata the `pData()` function is used to return a
`data.frame` containing subject-level variables of study.

```{r, eval=FALSE}
LomanNJ_2013_Hi.genefamilies_relab.stool() %>%
pData()
```

To access the data itself the `exprs()` function is used to return a long and
skinny `matrix` in which samples are column rather than rows.

```{r, eval=FALSE}
LomanNJ_2013_Hi.genefamilies_relab.stool() %>%
exprs()
```

# Taxonomy-Aware Analysis using `phyloseq`

The `phyloseq` bioconductor package provides a number of useful functions
related to taxonomic analysis and readily produces `ggPlot2` graphics.
`curatedMetagenomicData` provides 6 types of data; of these, only the taxonomic
profiles make sense to analyse as phylogenetic data. A convienence function,
`ExpressionSet2phyloseq()`, is provided within the `curatedMetagenomicData`
package to allow for coversion between the `ExpressionSet` class and the
`phyloseq` class; it is used as follows.

```{r, eval=FALSE}
LomanNJ_2013_Hi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq()
```

## Exploratory Data Analysis

Exploratory data analysis might begin through the examination of clade-specific
realative abundances contained in the OTU (Operational Taxanomic Unit) table. A
dataset containing taxanomic profiles is first converted to a `phyloseq` object
and then the `otu_table()` function is used to view the contents and structure
of the table.

```{r, eval=FALSE}
LomanNJ_2013_Mi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq() %>% 
otu_table() %>% 
head()
```

Similarly, the process of exploratory data analysis will likely include an
examination of study-related metadata in order to understand the type of
questions that can be asked of the data. To examine the structure and contents
of metadata a `phyloseq` object is created and then the `sample_data()` function
is used to view the table.

```{r, eval=FALSE}
LomanNJ_2013_Mi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq() %>% 
sample_data() %>% 
head()
```

As with the OTU and metadata tables, the structure and contents of the taxanomic
table should also be examined. This is done by converting an `ExpressionSet`
object to a `phyloseq` object and then using the `tax_table()` function to
expose the underlying taxanomic table. The table provides a complete listing of
every clade available within the dataset down to the strain level.

```{r, eval=FALSE}
LomanNJ_2013_Mi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq() %>%
tax_table() %>%
head()
```

## Subseting / Pruning

The names of phylogenetic ranks are easily examined using the `rank_names()`
function

```{r, eval=FALSE}
LomanNJ_2013_Mi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq() %>%
rank_names()
```

With knowledge of the rank names it becomes possible to filter 

```{r, eval=FALSE}
LomanNJ_2013_Mi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq() %>%
subset_taxa(!is.na(Species))
```

Even greater specificity

```{r, eval=FALSE}
LomanNJ_2013_Mi.metaphlan_bugs_list.stool() %>%
ExpressionSet2phyloseq() %>%
subset_taxa(!is.na(Strain))
```

## Advanced Pruning

## Taxonomy Heatmap

## Taxonomy Histogram

## Alpha Diversity Estimation

## Alpha Diversity Estimate Comparison

## Beta Diversity / Dissimilarity Clustering

## Beta Diversity / Dissimilarity Analysis
