<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Our Pipeline • curatedMetagenomicData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Our Pipeline">
<meta property="og:description" content="curatedMetagenomicData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">curatedMetagenomicData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.4.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/curatedMetagenomicData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/available-studies.html">Available Studies</a>
    </li>
    <li>
      <a href="../articles/our-pipeline.html">Our Pipeline</a>
    </li>
    <li>
      <a href="../articles/version-three.html">Version Three</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/curatedMetagenomicData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Our Pipeline</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/curatedMetagenomicData/blob/HEAD/vignettes/articles/our-pipeline.Rmd" class="external-link"><code>vignettes/articles/our-pipeline.Rmd</code></a></small>
      <div class="hidden name"><code>our-pipeline.Rmd</code></div>

    </div>

    
    
<p>To create the objects that are ultimately returned to users there are
a number of steps taken beyond what is contained in the
curatedMetagenomicData package – this document is intended to add some
clarity and transparency to the process. Roughly, the process is as
follows: a study is selected for inclusion, metadata for each sample in
the study is manually curated, raw data for each sample is downloaded
(usually from NCBI SRA), raw data is processed into the six data types
(gene families, marker abundance, marker presence, pathway abundance,
pathway coverage, and relative abundance) using <a href="https://github.com/biobakery/MetaPhlAn" class="external-link">MetaPhlAn3</a> and <a href="https://github.com/biobakery/humann" class="external-link">HUMAnN3</a>, processed data
for all samples in a study are combined into a matrix for each data
type, the six matrices are saved as R objects and uploaded to
ExperimentHub, and, finally, the matrices are used to construct
(Tree)SummarizedExperiment objects on the fly that are returned to
users.</p>
<div class="section level2">
<h2 id="curation">Curation<a class="anchor" aria-label="anchor" href="#curation"></a>
</h2>
<p>The curation process takes place on GitHub in the <a href="https://github.com/waldronlab/curatedMetagenomicDataCuration" class="external-link">curatedMetagenomicDataCuration</a>
repository. Sample metadata for every sample included in
curatedMetagenomicData is manually curated by humans and checked against
a standardized template for accuracy. The process is labor intensive and
information comes from three principle sources: journal articles (and
their supplementary materials), metadata included when samples were
uploaded to NCBI, and other publications referring to or augmenting the
original journal articles. Not only is explicit information (i.e. tables
that enumerate the values of variables for each sample) sought out, but
implicit information (e.g. “all the subjects in the study were
non-smokers”) is also included where it occurs in text. The manually
curated sample metadata is continuously updated with the advent of new
information (this is common when reanalyses find issues) and these
changes propagate from the curation repository to the R package
weekly.</p>
</div>
<div class="section level2">
<h2 id="processing">Processing<a class="anchor" aria-label="anchor" href="#processing"></a>
</h2>
<p>Raw data are processed using <a href="https://github.com/biobakery/MetaPhlAn" class="external-link">MetaPhlAn3</a> and <a href="https://github.com/biobakery/humann" class="external-link">HUMAnN3</a> and uploaded to a
Google Cloud Storage bucket as zipped tar file. The tar file contains
directories (one for each data type) of tsv files (one for each sample)
that are used as the input of R pipeline.</p>
</div>
<div class="section level2">
<h2 id="r-pipeline">R Pipeline<a class="anchor" aria-label="anchor" href="#r-pipeline"></a>
</h2>
<p>The collection of tsv files are not obviously or immediately useful –
there is an pipeline written in R to process these outputs into relevant
matrix objects for user consumption. The <a href="https://github.com/waldronlab/curatedMetagenomicDataPipeline" class="external-link">curatedMetagenomicDataPipeline</a>
repository on GitHub details the methods that are used to construct and
save the six matrices for each study that are uploaded to ExperimentHub.
The <code>gene_families</code> data type is large and stored as a sparse
matrix, while the other data types are store as standard matrix objects.
The upload to ExperimentHub is ultimately done using the aws-cli with
credentials from Bioconductor specific to the curatedMetagenomicData
package.</p>
<p>As for why we construct matrix objects: the process of combining the
samples into a matrix can take hours even on heavy-duty equipment (many
cores, many GB of RAM). This is both the advantage and crux of
curatedMetagenomicData – subseting the matrix takes mere seconds and
simple laptops are powerful enough; however, the ability to download
many single samples across studies is lost. We accomplish the later in
the R package by matrix subsetting and merging in the
<code><a href="../reference/returnSamples.html">returnSamples()</a></code> method. All alternatives considered, this
is the optimal trade-off, but we are working towards something more
efficient in the future – stay tuned!</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Lucas schiffer, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
